e2e-tests/utils.js

```diff
@@ -0,0 +1,44 @@
+const { Keyring } = require("@polkadot/api");
+const { Compact } = require("@polkadot/types");
+const { hexToU8a, u8aToHex } = require('@polkadot/util');
+const { cryptoWaitReady } = require('@polkadot/util-crypto');
+class Utils {
+    static TRANSACTION_LENGTH = 145;
+
+    /**
+     * Get Keyring pair prom private key
+     * @param {*} privateKey 
+     */
+    static async getKeyringPair(privateKey){
+        const keyring = new Keyring({ type: 'sr25519' });
+        await cryptoWaitReady();
+        const pair = keyring.addFromSeed(hexToU8a(privateKey));
+        return pair;
+    }
+
+    /**
+     * Sign the transaction with Keyring pair
+     * @param {*} from 
+     * @param {*} to 
+     * @param {*} amount 
+     * @param {*} nonce 
+     */
+    static async signTransaction(from, to, amount, nonce){
+        const len = Compact.encodeU8a(Utils.TRANSACTION_LENGTH);
+        const tx = {
+            from: u8aToHex(from.publicKey),
+            to,
+            amount,
+            nonce
+        };
+        let concatStr = "0x";
+        Object.values(tx).forEach(value => {
+            concatStr += value.slice(2);
```

Comment:
Instead of slicing directly `2`, I think that you can add additional util function `saveRemove0x` that receives a string and checks whether it starts with `0x` if yes, slices it, otherwise it does not do anything